relnotes/reverse.feature.md

```diff
@@ -0,0 +1,5 @@
+* `ocean.core.array.Iteration`
+
+  New module containing single function for now - `reverse`, which allows to
+  iterate given slice in reverse order without any allocations (similar to
+  built-in `foreach_reverse`).
```

Comment:
I agree with @gavin-norman-sociomantic.
`foreach_reverse` is clumsy-pragmatic: It works only with a subset of objects one can iterate over, support for `foreach` doesn't imply `foreach_reverse`, and it's probably problematic with ranges. But there are cases where `foreach_reverse` is applicable, most importantly arrays. In these cases it is simply useful. If `foreach_reverse` wouldn't exist I wouldn't request it, but given that it does exist and works perfectly fine for arrays I don't think it makes sense to boycot and work around it.